02_bubble/pfannerj/main.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var mainout io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintf(mainout, "Bubble sort starting...\n")
+	fmt.Fprintln(mainout, bubble([]int{3, 2, 1, 5}))
+	fmt.Fprintf(mainout, "Bubble sort completed...\n")
+	fmt.Fprintf(mainout, "Insertion sort starting...\n")
+	fmt.Fprintln(mainout, insertion([]int{3, 2, 5, 1}))
+	fmt.Fprintf(mainout, "Insertion sort completed...\n")
+}
+
+func bubble(s []int) []int {
+	sCopy := make([]int, len(s))
+	copy(sCopy, s)
```

Comment:
Hi Judy, please read the article below, as it gives you better slice copying without side-affects.
https://github.com/go101/go101/wiki/How-to-perfectly-clone-a-slice%3F

Reply:
ok, thx

Reply:
I spoke to Dan and he's recommended that I leave the slice copy code as it is for this solution...

Reply:
Yes, I changed mine too, so that it doesn't mutate. Please read the article below on the best solution to clone a slice:
https://github.com/go101/go101/wiki/How-to-perfectly-clone-a-slice%3F